### PR TITLE
Log clearer messages when the collector responds with UNIMPLEMENTED

### LINKED ISSUE #2571: Improve error message for connecting to collector without metrics endpoint
In open-telemetry/opentelemetry-java-instrumentation#1111 we found that all of our engineers misunderstood the error message for when connecting to a collector without a metrics endpoint. Given we default to tracing and metrics being enabled, I think this will be pretty common for users to run into. We should make the error message better.

### PR DESCRIPTION
**Pull request by jkwatson**
Resolves #2571

### CODE REVIEW

**Review by kenfinnigan — APPROVED** [review]

File: exporters/otlp/metrics/src/main/java/io/opentelemetry/exporter/otlp/metrics/OtlpGrpcMetricExporter.java [hunk 1]
```diff
@@ -78,7 +78,19 @@ public void onSuccess(@Nullable ExportMetricsServiceResponse response) {
 
           @Override
           public void onFailure(Throwable t) {
-            logger.log(Level.WARNING, "Failed to export metrics", t);
+            if (t.getMessage().contains("UNIMPLEMENTED")) {
```
Comment (anuraaga) [inline exporters/otlp/metrics/src/main/java/io/opentelemetry/exporter/otlp/metrics/OtlpGrpcMetricExporter.java:81]: I think you can use `Status.fromThrowable` for more structured access. Is it also a good time to add a message for UNAVAILABLE, the most common exception we get asked about now adays?

**Review by anuraaga — COMMENTED** [review]
Let's add some unit tests

File: exporters/otlp/metrics/src/main/java/io/opentelemetry/exporter/otlp/metrics/OtlpGrpcMetricExporter.java (hunk 1 above)
Comment (jkwatson) [reply exporters/otlp/metrics/src/main/java/io/opentelemetry/exporter/otlp/metrics/OtlpGrpcMetricExporter.java:81]: when does UNAVAILABLE happen? Very happy to add it, if I know a good message to provide.

Comment (anuraaga) [reply exporters/otlp/metrics/src/main/java/io/opentelemetry/exporter/otlp/metrics/OtlpGrpcMetricExporter.java:81]: It's generally a connectivity issue. The collector being down, the endpoint being set wrong, network config blocking the requests. A message along the lines of making sure collector is running and reachable by network

Comment (jkwatson) [reply exporters/otlp/metrics/src/main/java/io/opentelemetry/exporter/otlp/metrics/OtlpGrpcMetricExporter.java]: ok, check it out now.

**Comment by jkwatson** [comment]
> Let's add some unit tests

done.

**Comment by jkwatson** [comment]
@anuraaga anything else to address here?

File: exporters/otlp/metrics/src/main/java/io/opentelemetry/exporter/otlp/metrics/OtlpGrpcMetricExporter.java [hunk 2]
```diff
@@ -78,7 +80,35 @@ public void onSuccess(@Nullable ExportMetricsServiceResponse response) {
 
           @Override
           public void onFailure(Throwable t) {
-            logger.log(Level.WARNING, "Failed to export metrics", t);
+            if (t instanceof StatusRuntimeException) {
```
Comment (anuraaga) [inline exporters/otlp/metrics/src/main/java/io/opentelemetry/exporter/otlp/metrics/OtlpGrpcMetricExporter.java:83]: Can't you use fromThrowable? I think it handles both runtime and checked versions.

https://grpc.github.io/grpc-java/javadoc/io/grpc/Status.html#fromThrowable-java.lang.Throwable-

Comment (jkwatson) [reply exporters/otlp/metrics/src/main/java/io/opentelemetry/exporter/otlp/metrics/OtlpGrpcMetricExporter.java:83]: aha! I missed that. I'll update (although functionally it'll be the same, I think)

Comment (jkwatson) [reply exporters/otlp/metrics/src/main/java/io/opentelemetry/exporter/otlp/metrics/OtlpGrpcMetricExporter.java]: done

**Comment by jkwatson** [comment]
@anuraaga anything further you want on this PR?

**Review by anuraaga — APPROVED** [review]

**Pull request merged**